### ISSUE TITLE
Flexible objective selection

### ISSUE THREAD

**Issue by joroeder** [oemof/oemof-solph#862]
Right now, the flows can be weighted with any factor by additional keywords, e.g. as `emission_factor` for e.g. adding emission constraints.

It would be nice, if the user can select easily the objective, and optimise e.g. for any of the constraints, such as `emission_factor_constraint`.

The work-around would be to use the emission factor values as `variable_costs`, but I am wondering if there is an easier way to do this.

Do you have an idea how this could be implemented?

**Comment by p-snft** [comment]
Actually, I wouldn't call that a workaround. Instead, I interpret the `variable_costs` as generalised costs as in “cost function” (in a mathematical meaning).

**Comment by joroeder** [comment]
Yes, generally I also see it like that ... however, I would like to know at the same time know the total costs, if I use the emission values as `variable_costs`. My question is maybe more on, is it possible to use any constraint function as objective on oemof.solph level? Or do I need to use the pyomo level for that, as this would require to much effort implementing this in oemof.solph?

**Comment by p-snft** [comment]
It should be possible to add terms to the objective function in Pyomo, so this could be exposed also at solph level. Is this an urgent topic or should we discuss that at the dev meeting?

**Comment by joroeder** [comment]
No, it is not urgent, more a nice-to-have feature :) Yes, we could discuss this at the next meeting.
